kvbc/src/v4blockchain/detail/blockchain.cpp

```diff
@@ -64,6 +64,43 @@ BlockId Blockchain::addBlock(const concord::kvbc::categorization::Updates& categ
   return id;
 }
 
+// Delete up to until not including until,
+// returns the last block id that was deleted.
+BlockId Blockchain::deleteBlocksUntil(BlockId until) {
+  ConcordAssertGT(genesis_block_id_, INVALID_BLOCK_ID);
+  ConcordAssertLT(genesis_block_id_, until);
+  // We have a single block on the chain
+  if (last_reachable_block_id_ == genesis_block_id_) return genesis_block_id_ - 1;
+  // We don't want to erase all the blockchain
+  const auto last_deleted_block = std::min(last_reachable_block_id_.load() - 1, until - 1);
```

Comment:
This makes it safe to delete a blockchain with any block id. 
so to delete the complete blockchain we can use the following functions:
void deleteWholeBlockChain() {
    deleteBlocksUntil(MAX_BLOCK_ID);
    deleteLastReachableBlock();
    deleteGenesisBlock();
}

Asking the above, just for my understanding. We can add a testcase with the above.

Reply:
both deleteLastReachableBlock and deleteGenesisBlock have
```ConcordAssertLT(genesis_block_id_, last_reachable_block_id_);```
So it shouldn't be possible to erase the whole blockchain.
I'll add a test case for it